tests/testsuite/cargo_remove/gc_patch/out/Cargo.toml

```diff
@@ -5,5 +5,8 @@ members = [ "my-member" ]
 name = "my-project"
 version = "0.1.0"
 
+[patch."[ROOTURL]/bar1"]
+bar = { git = "[ROOTURL]/bar2" }
+
 [patch.crates-io]
 bar = { git = "[ROOTURL]/bar2" }
```

Comment:
Something doesn't seem to be working right

This test has
- my-project depend on bar from git and does a `cargo remove` on it
- my-member depends on bar from crates.io

The crates-io patch should remain because that is untouched but the git patch should be removed.

Reply:
I spent some time here trying to figure out why this was the case as well. I inspected the sandbox for the test and I had noticed that the patch wasn't being set as unused evidenced by the lack of emitted warning and the Cargo.lock file didn't have any `patch.unused` sections.

I'll look into this a bit more.

Reply:
Ok so the issue is that both patch dependency have the same PackageId. Since one of the patches is used and both patches from the same PackageId, neither of the patches show up as unused. There doesn't really appear to be an easy way to determine if a specific dependency was patched so it looks like to reliably know if a patch is unused, it'd have to be populated during graph generation.

Do we support this use case where multiple dependencies with different sources are overridden by the same patch dependency?

Reply:
Realized that I wasn't writing to the manifest so I updated it. Also, I'm taking the Dependency Url into account when looking through the list of unused_patches just in case there are multiple with the same name